lib/proxy/helpers.rb

```diff
@@ -40,6 +40,25 @@ def https_cert_cn
     end
   end
 
+  # parses the body as json and returns a hash of the body
+  # returns empty hash if there is a json parse error or body is empty
+  def parse_json_body
+    json_data = {}
+    if request.env["CONTENT_TYPE"] == "application/json"
+      begin
+        body_parameters = request.body.read
+        unless body_parameters.nil? || body_parameters.empty?
+          json_data = JSON.parse(body_parameters)
+        end
+      rescue => e
+        logger.debug("Invalid JSON content in body of request: #{e.message}")
```

Comment:
The idea was to swallow the error and assume the user would view the logs should something go wrong with the request.  But it could also be useful to return 400 as well.  I'll go ahead and change to 400 since I was on the fence about this already.